create.ps1

```diff
@@ -1,258 +1,449 @@
+    $account.PsObject.Properties | ForEach-Object {
+        # Remove properties with null-values
+        if ($_.Value -eq $null) {
+            $account.PsObject.Properties.Remove("$($_.Name)")
         }
     }
+    # Convert the properties of account object containing "TRUE" or "FALSE" to boolean 
+    $account = Convert-StringToBoolean $account
+    #endRegion account
 
-    #region Correlation
-    if ($actionContext.CorrelationConfiguration.Enabled) {
-        $CorrelationField = $actionContext.CorrelationConfiguration.accountField
-        $CorrelationValue = $actionContext.CorrelationConfiguration.PersonFieldValue
+    #region Verify correlation configuration and properties
+    $actionMessage = "verifying correlation configuration and properties"
 
-        if ($Null -eq $CorrelationField -or $Null -eq $CorrelationValue) {
+    if ($actionContext.CorrelationConfiguration.Enabled -eq $true) {
+        if ([string]::IsNullOrEmpty($correlationField)) {
             throw "Correlation is enabled but not configured correctly."
         }
-
-        #  Write logic here that checks if the account can be correlated in the target system
-        $SplatParams = @{
-            Uri    = "$($actionContext.Configuration.BaseUrl)/Users"
-            Method = "Get"
-            Body   = @{
-                filter = "$($CorrelationField) eq '$($CorrelationValue)'"
-            }
+        
+        if ([string]::IsNullOrEmpty($correlationValue)) {
+            throw "The correlation value for [$correlationField] is empty. This is likely a mapping issue."
         }
-        $CorrelatedAccount = Invoke-RestMethod @LisaRequest @SplatParams
-
-        if ($CorrelatedAccount.count -gt 1) {
-            throw "Multiple accounts found with filter: $($SplatParams.Body.filter)"
+    }
+    else {
+        Write-Warning "Correlation is disabled."
+    }
+    #endregion Verify correlation configuration and properties
+    
+    #region Create access token
+    $actionMessage = "creating access token"
+    
+    $createAccessTokenBody = @{
+        grant_type    = "client_credentials"
+        client_id     = $actionContext.Configuration.EntraIDAppId
+        client_secret = $actionContext.Configuration.EntraIDAppSecret
+        scope         = $actionContext.Configuration.KPNMWPScope
+    }
+    
+    $createAccessTokenSplatParams = @{
+        Uri             = "https://login.microsoftonline.com/$($actionContext.Configuration.EntraIDTenantID)/oauth2/v2.0/token/"
+        Headers         = $headers
+        Method          = "POST"
+        ContentType     = "application/x-www-form-urlencoded"
+        UseBasicParsing = $true
+        Body            = $createAccessTokenBody
+        Verbose         = $false
+        ErrorAction     = "Stop"
+    }
+    
+    $createAccessTokenResonse = Invoke-RestMethod @createAccessTokenSplatParams
+    
+    Write-Verbose "Created access token. Result: $($createAccessTokenResonse | ConvertTo-Json)"
```

Comment:
Misschien kunnen we ipv het volledige token object naar de logging schrijven, slechts de 'expiresIn' of 'scope' wegschrijven?

Reply:
Eens, de gehele token of headers in logging wegschrijven is sowieso niet gewenst.